[SinhaVedant] iP

SinhaVedant opened the pull request:
# Sivraj
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Sivraj frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use.

All you need to do is, 
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

SinhaVedant committed:
Level-0

SinhaVedant committed:
Level-1

SinhaVedant committed:
Level-2

SinhaVedant committed:
Level-3

SinhaVedant committed:
Level-4

SinhaVedant committed:
A-TextUiTesting

SinhaVedant committed:
Level-5

SinhaVedant committed:
Level-6

SinhaVedant committed:
Improve text UI

SinhaVedant committed:
Level-7

SinhaVedant committed:
Merge branch 'branch-Level-7'

SinhaVedant committed:
Level-8

SinhaVedant committed:
Merge branch 'branch-Level-8'

SinhaVedant committed:
A-MoreOOP

SinhaVedant committed:
A-Packages

SinhaVedant committed:
A-Gradle

SinhaVedant committed:
A-JUnit

SinhaVedant committed:
A-Jar

SinhaVedant committed:
A-JavaDoc

SinhaVedant committed:
A-CodingStandard

SinhaVedant committed:
Level-9

SinhaVedant committed:
Merge branch 'branch-A-JavaDoc'

SinhaVedant committed:
Resolve A-CodingStandard Conflict

SinhaVedant committed:
Resolve Level-9 Conflict

SinhaVedant committed:
Level-10

SinhaVedant committed:
Merge branch 'branch-Level-10'

darrentfy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,218 @@
+                    }
+                    String description = echo.substring(5).trim();
+                    taskList.add(new ToDo(description, 'T'));
+
+                        System.out.println(dashLine);
+                        System.out.println("     Got it. I've added this task:");
+                        System.out.println("         " + taskList.get(taskList.size() - 1));
+                        System.out.println("     Now you have " + taskList.size() + " tasks in the list.");
+                        System.out.println(dashLine);
```
Some unnecessary indentation here

darrentfy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,218 @@
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        switch (taskType) {
+            case 'T':
+                ToDo toDoTask = new ToDo(description, taskType);
+                if (isDone) {
+                    toDoTask.markAsDone();
+                } else {
+                    toDoTask.markAsNotDone();
+                }
+                return toDoTask;
+
```
```suggestion
        switch (taskType) {
        case 'T':
```
If i'm not wrong, the 'case' line should not have indentation

darrentfy reviewed (commented):
Apart from a couple of minor problems with regards to coding style, your code is easy to read overall.

kokrui commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run duke.Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
I believe this is a typo that might've been introduced by IntelliJ's auto-refactoring process?

kokrui commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,218 @@
+import java.io.IOException;
+
+
+public class Duke {
```
It would be good to add header comments to public classes

kokrui reviewed (commented):
Overall a reasonable implementation, with a few coding standard nits

SinhaVedant committed:
Add full commit message

Our previous commits did not have a full commit message.

A full commit message is required for this tag.

Let's add a full commit message.

SinhaVedant committed:
Use assert feature

Our previous code did not use assert feature.

Using assert feature can help us to document important assumptions that
should hold at various points in the code.

Let's use the assert feature.

SinhaVedant committed:
Improve code quality

SinhaVedant committed:
Merge pull request #1 from SinhaVedant/branch-A-Assertions

Use assert feature

SinhaVedant committed:
Resolve merge conflict

SinhaVedant committed:
Merge pull request #2 from SinhaVedant/branch-A-CodeQuality

Improve code quality

SinhaVedant committed:
Commit changes in duke txt file

SinhaVedant committed:
Merge branch 'master' into branch-A-CodeQuality

SinhaVedant committed:
Add a brief User Guide (UG)

SinhaVedant committed:
Merge branch 'master' into branch-Level-8

SinhaVedant committed:
Update deadline date format

SinhaVedant committed:
Merge branch 'branch-Level-8'

SinhaVedant committed:
Test the latest update in master branch

SinhaVedant committed:
Update Ui.java file

SinhaVedant committed:
Update Storage.java file

SinhaVedant committed:
Update parser class

SinhaVedant committed:
Implement better OOP

SinhaVedant committed:
Merge branch 'branch-A-MoreOOP'

SinhaVedant committed:
Merge branch 'add-gradle-support'

SinhaVedant committed:
Add test-cases for Storage and Parser classes

SinhaVedant committed:
Create a jar file

SinhaVedant committed:
Complete tutorial of GUI

SinhaVedant committed:
Complete GUI for the project

SinhaVedant committed:
Merge branch 'GUI'

SinhaVedant committed:
Add assertions to the code

SinhaVedant committed:
Add JavaDoc to some classes and methods

SinhaVedant committed:
Add screenshot of ip project

SinhaVedant committed:
Update the user guide

SinhaVedant committed:
Create final jar file

SinhaVedant committed:
Set mainClass to duke.Launcher

SinhaVedant committed:
Improve code quality

SinhaVedant committed:
Merge branch 'improve'

SinhaVedant committed:
Improve Ui

SinhaVedant committed:
Update Storage class to create a new file

SinhaVedant committed:
Merge branch 'improve'

SinhaVedant committed:
Update JavaDocs

SinhaVedant committed:
Improve GUI of Sivraj

SinhaVedant committed:
Merge branch 'gui2'

SinhaVedant committed:
Resolve merge conflict

SinhaVedant committed:
Merge pull request #4 from SinhaVedant/branch-A-CodeQuality

Branch a code quality

SinhaVedant committed:
Improve JavaDoc
